src/apiserver/api_server_test.cc

```diff
@@ -145,6 +145,45 @@ TEST_F(APIServerTest, jsonFormat) {
     ASSERT_EQ(butil::rapidjson::kNullType, arr[6].GetType());
 }
 
+TEST_F(APIServerTest, query) {
+    const auto env = APIServerTestEnv::Instance();
+
+    std::string ddl = "create table demo (c1 int, c2 string);";
+    hybridse::sdk::Status status;
+    ASSERT_TRUE(env->cluster_remote->ExecuteDDL(env->db, ddl, &status)) << "fail to create table";
+
+    std::string insert_sql = "insert into demo values (1, \"bb\");";
+    ASSERT_TRUE(env->cluster_sdk->Refresh());
+    ASSERT_TRUE(env->cluster_remote->ExecuteInsert(env->db, insert_sql, &status));
+
+    {
+        brpc::Controller cntl;
+        cntl.http_request().set_method(brpc::HTTP_METHOD_POST);
+        cntl.http_request().uri() = "http://127.0.0.1:8010/dbs/" + env->db;
+        cntl.request_attachment().append(R"({
+            "sql": "select c1, c2 from demo;", "mode": "offsync"
```

Comment:
this require ` "mode": "online"` field, api server default to `offline` mode

Reply:
addressed